### PR TITLE
Refactor and clean up RpcUtils

### PR DESCRIPTION
**Pull request by ggezer**

### CODE REVIEW

**Comment by ggezer** [comment]
@bf8086 PTAL. Thanks!

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 1]
```diff
@@ -186,44 +73,16 @@ private RpcUtils() {} // prevent instantiation
...
-              String.format(description, args), e);
-        }
-      }
-      responseObserver.onError(AlluxioStatusException.fromIOException(e).toGrpcStatusException());
-    } catch (RuntimeException e) {
-      logger.error("Exit (Error): {}: {}", methodName, String.format(description, args), e);
-      MetricsSystem.counter(getQualifiedFailureMetricName(methodName)).inc();
-      responseObserver.onError(new InternalException(e).toGrpcStatusException());
+          .onNext(callAndReturn(logger, callable, methodName, failureOk, description, args));
+      responseObserver.onCompleted();
```
Comment (bf8086) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: Move this line out of the try-catch to avoid error due to calling onError on closed call?

Comment (ggezer) [reply core/server/common/src/main/java/alluxio/RpcUtils.java]: Did some more hardening.

**Comment by ggezer** [comment]
Jenkins, test this please

**Comment by ggezer** [comment]
Jenkins, test this please

**Comment by ggezer** [comment]
Jenkins, test this please

**Comment by ggezer** [comment]
@madanadit PTAL.

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 2]
```diff
@@ -25,130 +25,17 @@
...
  * Utilities for handling server RPC calls.
  *
- * There are three types of RPC calls: 1. RPCs that only throw AlluxioException 2. RPCs that throw
- * AlluxioException and IOException 3. Streaming RPCs
- *
- * For each of these, there are two types of methods 1. call(callable) - for internal methods,
- * executes the method without any logging or metrics 2. call(logger, callable, method name,
- * failureOk, method description, arguments...) - for client initiated methods, executes the method
- * with logging and metrics. If failureOk is set, non-fatal errors will only be logged at the DEBUG
+ * There are three types of RPC calls: 1. RPCs that only throw IOException 2. Streaming RPCs
```
Comment (madanadit) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: nit: two types?

**Comment by madanadit** [comment]
alluxio-bot, merge this please.

**Pull request merged**